src/applications/personalization/profile/actions/directDeposit.js

```diff
@@ -98,16 +98,56 @@ export function saveDirectDeposit(
 
     const response = await getData(DIRECT_DEPOSIT_API_ENDPOINT, {
       method: 'PUT',
+      headers: {
+        'Content-Type': 'application/json',
+      },
       body: JSON.stringify(data),
     });
 
-    // error handling for the response
-    if (response.error || response.errors) {
-      recordDirectDepositEvent({
+    // for client side errors
+    // check if response is an instance of an error TypeError or any other error
+    if (response instanceof Error) {
+      const event = {
+        method: 'PUT',
```

Comment:
Added some client side error handling for if the user loses internet or another client side error occurs.